README.md

```diff
@@ -1,2 +1,285 @@
+`"{{workflow.inputs.parameters.file_path}}"` and Queenbee will map the value of
+`file_path` parameter to this placeholder.
+
+Inputs can be from two different types: `parameters` and `artifacts`. Use `artifacts` for
+*files* and *folder* inputs. The content of these `artifacts` will be copied to target
+location before executing the workflow. For all the other `inputs` use `parameters`.
+
+Here is an example `inputs` for a daylight simulation workflow. You can put a default
+value for inputs or it can be passed to workflow using an input YAML / JSON file.
+
+```yaml
+
+  inputs:
+    parameters:
+    # this is just the path to target folder not the content itself. It can be referenced
+    # as {{workflow.parameters.folder}}
+    - name: folder
+      value: /mnt/c/ladybug/untitled
+    artifacts:
+    - name: sky_file
+      value: full_path_to_sky_file.sky
+    - name: material_file
+      value: full_path_to_material_file.mat
+    - name: geometry_file
+      value: full_path_to_geometry_file.rad
+
+```
+
+## 2. operators
+
+`Operators` include the requirements for running `templates` locally or using a Docker
+image. This is an operator for running arbitrary Radiance commands.
+
+```yaml
+
+operators:
+  - name: radiance-operator
+    inputs:
+      parameters:
+        - name: command
+    container:
+      name: radiance52
+      image: ladybugtools/radiance:5.2
+      command:
+        - /bin/sh
+        - -c
+      args:
+        - '{{ inputs.parameters.command }}'
+    requirements:
+      app:
+        - name: radiance
+          version: ">=5.2"
+          command: rtrace -version  # command to get the version
+          pattern: r'\d+\.\d+'  # regex pattern to extract the version from command output
+
+```
+
+Here is another one for running a `template` for running `ladybug` Python scripts.
+This `operator` will run on Windows, Linux and Mac platforms. You can exclude a platform
+by removing it from the list.
+
+```yaml
+
+  - name: ladybug-operator
+    inputs:
+      parameters:
+        - name: script
+    container:
+      name: ladybug
+      image: ladybugtools/ladybug:latest
+      command:
+        - python
+      args:
+        - '{{ inputs.parameters.script }}'
+    requirements:
+      pip:
+        - name: lbt-ladybug
+          version: ">=0.4.3"
+      language:
+        - name: python
+          version: ">=3.6"
+      platform:
+       - Windows
+       - Linux
+       - Mac
+
+```
+
+You can also import the operator from a `YAML` or `JSON` file using the `import_from`
+key.
+
+```yaml
+
+  - name: ladybug-operator
+    inputs:
+      parameters:
+        - name: script
+    import_from: /operators/ladybug_operator.yaml
+
+```
+
+## 3. templates
+
+Templates are a discrete reusable unit of code that can be executed separately or as part
+of a workflow. Queenbee supports 4 type of objects as templates:
+
+- `module` : A single module. Each module uses an operator from operators section.
```

Comment:
Is a module essentially 1 workflow task such as `create skyfile` or `run gridbased simulation`?

Reply:
Yes. `module` is the smallest executable unit of templates. I used to call them `template` but it was confusing as all of these different types are supposed to be used as templates.

Reply:
added examples for modules to the document.